### PR TITLE
fix(Packaging): Fix support of the artifact S3 uri with region

### PR DESCRIPTION
**Pull request by zachwhaley**
#9380 Added support for most S3 bucket URIs, but not ones with region in their format, e.g. any bucket outside of us-east-1.

Closes: #9403

### CODE REVIEW

File: lib/plugins/aws/utils/parse-s3-uri.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
 'use strict';
 
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html
 const patterns = [
   // S3 URI. Ex: s3://bucket/path/to/artifact.zip
-  new RegExp('^s3://([^/]+)/(.+)'),
+  new RegExp('^s3://(?<bucket>[^/]+)/(?<key>.+)'),
```
Comment (bishtawi) [inline lib/plugins/aws/utils/parse-s3-uri.js:6]: I didnt realize you could label your matches. That is really cool.

File: lib/plugins/aws/utils/parse-s3-uri.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
 
-  // New style S3 URL. Ex: https://bucket.s3.amazonaws.com/path/to/artifact.zip
-  new RegExp('([^/]+)\\.s3\\.amazonaws\\.com/(.+)'),
+  // New style S3 URL. Ex: https://bucket.s3.REGION.amazonaws.com/path/to/artifact.zip
+  new RegExp('(?<bucket>[^/]+)\\.s3(\\.[\\w\\d-]+)?\\.amazonaws\\.com/(?<key>.+)'),
 
-  // Old style S3 URL. Ex: https://s3.amazonaws.com/bucket/path/to/artifact.zip
-  new RegExp('s3\\.amazonaws\\.com/([^/]+)/(.+)'),
+  // Old style S3 URL. Ex: https://s3.REGION.amazonaws.com/bucket/path/to/artifact.zip
+  new RegExp('s3(\\.[\\w\\d-]+)?\\.amazonaws\\.com/(?<bucket>[^/]+)/(?<key>.+)'),
```
Comment (bishtawi) [inline lib/plugins/aws/utils/parse-s3-uri.js:12]: For this regex and the one above, you should probably support urls with dashes separating the `s3` from the `region` as apparently `s3-region` is a valid url is some aws regions.

Ex: `https://my-bucket.s3-us-west-2.amazonaws.com`

So `s3([.-][\\w\\d-]+)?`.

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @zachwhaley - it looks great :raised_hands: Before we merge, it looks like there are minor issues with formatting - could you please take a look at it? After that we should be good to go :rocket:

**Review by pgrzesik — APPROVED** [review]
Thank you @zachwhaley, looks great :+1:

**Pull request merged**